### PR TITLE
Added warning for VHD creation and updated samples

### PR DESCRIPTION
**Pull request by harijayms**
Creating VHDs is a older way creating custom images in Azure. Azure is recommending using managed Images. This PR provides user with a Warning that VHD disk path is going to get deprecated and updates the samples to reflect the same 

Please include tests. Check out these examples:

Tested the case for both VHD and Managed image to check error messages. The error message doesn't not fail the build. 

Also validated all samples as well.

### CODE REVIEW

**Comment by harijayms** [comment]
@boumenot @paulmey can you please take a look

File: examples/azure/centos.json [hunk 1]
```diff
@@ -2,25 +2,21 @@
   "variables": {
     "client_id": "{{env `ARM_CLIENT_ID`}}",
     "client_secret": "{{env `ARM_CLIENT_SECRET`}}",
-    "resource_group": "{{env `ARM_RESOURCE_GROUP`}}",
-    "storage_account": "{{env `ARM_STORAGE_ACCOUNT`}}",
     "subscription_id": "{{env `ARM_SUBSCRIPTION_ID`}}",
     "tenant_id": "{{env `ARM_TENANT_ID`}}",
     "ssh_user": "centos",
-    "ssh_pass": null
+    "ssh_pass": "{{env `ARM_SSH_PASS`}}"
```
[the remark this answers is not in the document]
Comment (harijayms) [reply examples/azure/centos.json]: Sure I can change it to public keys, passwords are its easier from a sample point of view and copy paste.

[the remark this answers is not in the document]
Comment (paulmey) [reply examples/azure/centos.json]: I understand your point, but the user never uses this password. It's only used by Packer's provisioners...

File: examples/azure/debian.json [hunk 2]
```diff
@@ -2,11 +2,9 @@
   "variables": {
```
Comment (boumenot) [inline examples/azure/debian.json]: Please update the version to 9.

File: examples/azure/centos.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (harijayms) [reply examples/azure/centos.json]: Will address this in a separate PR

File: examples/azure/debian.json (hunk 2 above)
Comment (harijayms) [reply examples/azure/debian.json]: done.

**Comment by harijayms** [comment]
@boumenot  ball is in your court

File: examples/azure/centos.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (boumenot) [reply examples/azure/centos.json]: This is N/A.  This is for **some** distros that requires a password when you sudo. The password is not used to connect to the VM. The Azure builder has always used public key authentication when communicating with VMs.

**Pull request merged**